spec/shared/json_convertible_spec.rb

```diff
@@ -77,6 +77,16 @@ def self.map_enumerable_type(enumerable_property_name: nil, current_json_object:
   end
 end
 
+class MockJSONConvertibleWithRequiredParams
+  include FastlaneCI::JSONConvertible
+
+  attr_reader :one_attribute
+
+  def initialize(one_attribute:)
+    @one_attribute = one_attribute
+  end
+end
+
 module FastlaneCI
   describe JSONConvertible do
```

Comment:
I think we need a few more specs to be safe:
1. flex the various failure cases (how many failure cases are there?)
2. objects that have a mix of required and non-required params
3. objects that have multiple required params
4. objects that have a mix of multiple required and multiple non-required params

Reply:
Seems legit to me, will try to get this done for the weekend!